File: tools/js/eat_cli.ts
// eat_cli.ts
// Simple CLI for EAT <-> JSON conversion.

import * as fs from "fs";
import * as path from "path";
import { parseEAT, stringifyEAT } from "./eat_parser";

function printUsage() {
  console.log("Usage:");
  console.log("  eat-cli parse <file.eat>");
  console.log("  eat-cli stringify <file.json>");
  process.exit(1);
}

async function main() {
  const [, , cmd, file] = process.argv;
  if (!cmd || !file) {
    printUsage();
  }

  if (!fs.existsSync(file)) {
    console.error(`File not found: ${file}`);
    process.exit(1);
  }

  if (cmd === "parse") {
    const text = fs.readFileSync(file, "utf8");
    const doc = parseEAT(text);
    console.log(JSON.stringify(doc, null, 2));
  } else if (cmd === "stringify") {
    const text = fs.readFileSync(file, "utf8");
    const json = JSON.parse(text);
    const eatText = stringifyEAT(json);
    console.log(eatText);
  } else {
    printUsage();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: eat_cli.py
"""eat_cli.py
CLI tool for EAT <-> JSON conversion.

Usage:
  python eat_cli.py parse file.eat
  python eat_cli.py stringify file.json
"""
import sys
import json
from pathlib import Path
from eat_parser import parse_eat, stringify_eat


def main() -> None:
    if len(sys.argv) != 3:
        print(__doc__)
        raise SystemExit(1)

    cmd, file = sys.argv[1], Path(sys.argv[2])
    if not file.exists():
        print(f"File not found: {file}")
        raise SystemExit(1)

    if cmd == "parse":
        text = file.read_text(encoding="utf-8")
        doc = parse_eat(text)
        print(json.dumps(doc, indent=2, ensure_ascii=False))
    elif cmd == "stringify":
        text = file.read_text(encoding="utf-8")
        data = json.loads(text)
        eat_text = stringify_eat(data)
        print(eat_text, end="")
    else:
        print(__doc__)
        raise SystemExit(1)


if __name__ == "__main__":
    main()
